src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/SystemRegenerateJsCitizenAosResponseLetterTask.java

```diff
@@ -0,0 +1,90 @@
+import org.elasticsearch.index.query.QueryBuilders;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.divorce.idam.IdamService;
+import uk.gov.hmcts.divorce.idam.User;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdConflictException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdManagementException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdSearchCaseException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdUpdateService;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static uk.gov.hmcts.divorce.systemupdate.event.SystemRegenerateJsCitizenAosResponseCoverLetter.SYSTEM_REGEN_JS_CITIZEN_AOS_RESPONSE_COVER_LETTER;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+/**
+ * SystemRegenerateJsCitizenAosResponseLetterTask introduced as a one off fix for NFDIV-4237.
+ * Any js cases sent an AoS pack to unrepresented citizens since Nov 2023, used the solicitor template for the cover letter which resulted
+ * in a blank address line.
+ * 'JS AoS Response pack cover letters' should be regenerated with address on the top and sent to the respective applicant.
+ */
+public class SystemRegenerateJsCitizenAosResponseLetterTask implements Runnable {
+
+    public static final String CASE_ID_CSV = "NFDIV-4237.csv";
+    public static final String SEARCH_ERROR = "SystemRegenerateJsCitizenAosResponseCoverLetterTask stopped after search error";
+    public static final String CONFLICT_ERROR =
+        "SystemRegenerateJsCitizenAosResponseCoverLetterTask stopping due to conflict with another running task";
+    public static final String FILE_READ_ERROR = "SystemRegenerateJsCitizenAosResponseCoverLetterTask stopped after file read error";
+    public static final String SUBMIT_EVENT_ERROR = "Submit event failed for case id: {}, continuing to next case";
+    public static final String DESERIALIZATION_ERROR = "Deserialization failed for case id: {}, continuing to next case";
+
+    private final CcdUpdateService ccdUpdateService;
+
+    private final CcdSearchService ccdSearchService;
+
+    private final IdamService idamService;
+
+    private final AuthTokenGenerator authTokenGenerator;
+
+    private final TaskHelper taskHelper;
+
+    @Override
+    public void run() {
+        log.info("SystemRegenerateJsCitizenAosResponseCoverLetterTask started");
+        try {
+            final User user = idamService.retrieveSystemUpdateUserDetails();
+            final String serviceAuth = authTokenGenerator.generate();
+
+            List<Long> caseIds = taskHelper.loadCaseIds(CASE_ID_CSV);
+            final BoolQueryBuilder query = boolQuery()
+                .filter(QueryBuilders.termsQuery("reference", caseIds));
```

Comment:
How about filtering by the `JsCitizenAosResponseLettersResent` flag here or checking the flag before triggering the task on line 67? 🤔 It would allow us to avoid [this logging](https://github.com/hmcts/nfdiv-case-api/pull/3935/files#diff-a44b64cc6ce5cb2549f04f52d49a57cbb0c4c8d2a6fcfbb15ac1f71da5279aefR47)

Reply:
I thought it would be useful to be able to confirm if a something was getting skipped (just in case).

Reply:
I suppose a bigger problem with filtering at the end is we could put the event in case history multiple times when it does nothing. The extra logging and wasted work of calling the event/task multiple times probably aren't an issue if it's a small migration

Reply:
It's sub 60 cases this time, so very small.  I'll have a think about it anyway.  Can probably get the best of both by moving the log entry.